src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptyforiteratorpad/Example2.java

```diff
@@ -0,0 +1,25 @@
+/*
+EmptyForIteratorPad
+option = space
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.emptyforiteratorpad;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+class Example2 {
+  Map<String, String> map = new HashMap<>();
+  void example() {
+    for (Iterator it = map.entrySet().iterator();  it.hasNext(););
+    // violation above '';' is not followed by whitespace.'
+    for (Iterator it = map.entrySet().iterator();  it.hasNext(); );
+
+    for (Iterator foo = map.entrySet().iterator();
+         foo.hasNext();
+         );
```

Comment:
Was `violation as there  is no whitespace after semicolon` (from the existing docs) a false positive here?

Reply:
I believe, yes. Even the description **talks about this specific case** and how there is no check happening there.
https://checkstyle.sourceforge.io/config_whitespace.html#EmptyForIteratorPad_Description
>  Checks the padding of an empty for iterator; that is whether a white space is required at an empty for iterator, or such white space is forbidden. No check occurs if there is a line wrap at the iterator, as in 
```java
for (Iterator foo = very.long.line.iterator();
      foo.hasNext();
     )
```

Even the tests have no violation there
https://github.com/checkstyle/checkstyle/blob/79d3c5536760d2645b0676c5a7f31da3bb0eed72/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptyforiteratorpad/InputEmptyForIteratorPad.java#L34-L37

Reply:
Ok, please make sure to leave a review comment on things like this in the future to save time.

Reply:
Will do.

Reply:
Just another good catch of automation, it will be not a rare problem we resolve in xdoc examples.
Keep in mind that all examples are created by students, they barely able understand what they are doing and how it works.